src/engine/renderer/Material.cpp

```diff
@@ -1979,9 +1989,67 @@ void MaterialSystem::RenderMaterials( const shaderSort_t fromSort, const shaderS
 				continue;
 			}
 
+			// Use stencil buffer to avoid rendering stuff over the skybox that would normally be culled by the BSP
+			if ( backEnd.viewParms.portalLevel == 0 ) {
+				glEnable( GL_STENCIL_TEST );
+				glStencilMask( 0xff );
+			}
+
+			glStencilFunc( GL_EQUAL, backEnd.viewParms.portalLevel, 0xff );
+			glStencilOp( GL_KEEP, GL_KEEP, GL_INCR );
+
+			glState.glStateBitsMask = 0;
```

Comment:
I think it was breaking otherwise with portals. Maybe not and I put there just in case because it was breaking elsewhere, but either way doesn't matter now because this way of rendering the skybox seems infeasible. Well, maybe I'll actually keep the stenciled skybrush part to reset depth, I'll need to look at it again later.